spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerSortPageQueryOptions.java

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core;
+
+import java.util.OptionalLong;
+
+import org.springframework.data.domain.Sort;
+import org.springframework.util.Assert;
+
+/**
+ * Encapsulates Cloud Spanner query options with sort and paging.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+public class SpannerSortPageQueryOptions extends SpannerQueryOptions {
```

Comment:
In Spring they use the term "Pageable" to include bot page and sort options.
So, maybe `SpannerPageableQueryOptions`.